.github/workflows/deploy.yml

```diff
@@ -1,68 +1,81 @@
-name: Auto release PR
+name: Deploy
 
 on:
-  push:
-    tags:
-      - '*'
-    branches:
-      - staging
   workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to deploy'
+        required: true
+        type: string
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        type: choice
+        options:
+          - staging
+          - production
```

Comment:
nit: instead of parameter here, we want separate workflows (`deploy-staging.yml` and `deploy-production.yml`). 

this way each has own run history, so its easier to inspect what is deployed where without clicking into each job to see parameter + it is harder to misclick and deploy to wrong place when following release checklist with link to specific workflow.

Reply:
Good call, I now split them into two. I made them both use a common reusable workflow to avoid duplication.